### DISCUSSION TITLE
What means of sending notification textDocument/didClose failed ?

### DISCUSSION THREAD

**Discussion by KamiC6238** [TypeFox/monaco-languageclient#457] [Q&A]
I want to reconnect lsp when destroy the old editor model and re-install the MonacoServices for the new editor model, but it got an error like the discussion title, and that error would leading the new model using the wrong lsp of the old model.

![image](https://user-images.githubusercontent.com/23523595/216936598-624ab121-1233-4647-9e6f-ec76c5f1ddef.png)

When the new editor model being set, I will dispose the pre-MonacoServices and re-install the new MonacoServices like that pic.
![image](https://user-images.githubusercontent.com/23523595/216937518-cc48e2d2-fffb-46d4-a7d2-2b977bc5a9df.png)

How do I sovle this problem?

**Comment by CGNonofr** [comment]
Can you provide the full errors?

Also, It's suspicious that you set the workspace folder to your model path, it expects a directory path/uri

**Reply by KamiC6238** [reply]
> Can you provide the full errors?
> 
> Also, It's suspicious that you set the workspace folder to your model path, it expects a directory path/uri

The full errors.
This is because the old model and the new model also using java as the language documentSelector.
![image](https://user-images.githubusercontent.com/23523595/217132417-81f09700-dcd0-4647-ae44-6094c6b098dd.png)

And if the old model using java and the new model using c++, it wolud show more errors:
![image](https://user-images.githubusercontent.com/23523595/217132960-dd62ff93-f6bd-40bc-934f-2d84972b5293.png)

Also, In the backend, the model path will be mapped to direcotry path, because we considered the model uri would be created before install MonacoServices, so we decided to send the model path from frontend and the backend mapped it to the directory path.

**Reply by CGNonofr** [reply]
It says that the didClose notification failed, what is the cause of it? (open the error) Is the connection still open at this point?

If you have the java intellisense on a c++ file, it means that you either:
- have wrongly set the document selector
- have the wrong language on the model
